### DISCUSSION TITLE
RunProdServer tries to find RunDevServer oddly?

### DISCUSSION THREAD

**Discussion by nomeata** [digitallyinduced/ihp#1456] [General]
I am experimenting with IHP now, with a small site for a local club – I’m too proud to not use a custom domain name, but the club is far too small to pay for the cloud tiers. So I guess I’ll run it on a personal server of mine, a Debian server with `nix` installed.

Nicely, IHP provides the nix-based deployment builds, so that should hopefully make things easy. But I didn't get very far:
```
$ nix-build 
…
[16 of 16] Compiling Main             ( Main.hs, build/RunUnoptimizedProdServer/Main.o, build/RunUnoptimizedProdServer/Main.dyn_o )
Linking build/bin/RunUnoptimizedProdServer ...
chmod +x Main.hs
rm -f build/bin/RunProdServer
ln -s `basename build/bin/RunUnoptimizedProdServer` build/bin/RunProdServer
installing
/nix/store/l6vmnyfs4vc0kb0kprr8mhzkwch6xnx4-app
$ /nix/store/l6vmnyfs4vc0kb0kprr8mhzkwch6xnx4-app/
bin/ lib/ 
$ /nix/store/l6vmnyfs4vc0kb0kprr8mhzkwch6xnx4-app/bin/RunProdServer
dirname: fehlender Operand
„dirname --help“ liefert weitere Informationen.
RunProdServerWithoutOptions: readCreateProcess: dirname $(which RunDevServer) (exit 1): failed
```

Why is `RunProdServer` trying to find `RunDevServer`?

Oh, and does anyone have a nixos module generator for IHP projects around? Seems to be a natural fit…

**Comment by mpscholten** [comment]
> Why is RunProdServer trying to find RunDevServer?

IHP uses the location of the binary to find the location of the IHP [`lib` directory](https://github.com/digitallyinduced/ihp/tree/master/lib/IHP).

If you switch the working directory to `$out/lib` the IHP app should be able to start :) 

You can find an example nix configuration here: https://github.com/digitallyinduced/ihp/discussions/1292#discussioncomment-1880247

**Reply by mpscholten** [reply]
I've created https://github.com/digitallyinduced/ihp/issues/1457 to improve this 👍
